Check latest version

Neriderc opened the pull request:
This pull request adds version checking and updates the changelog (which was quite out of date).

Original post:

@hartenthaler I've made the changes as you've suggested in #69, but I don't see any warnings about the version.

I have set the latest-version.txt to have 2.1.9 and set the CUSTOM_VERSION constant to 2.1.8. I notice in the admin panel the current version of the module now shows, however, I don't get a warning about there being a new version.

Do checks happen on a schedule or should it work instantly?

I've created this pull request so you can see the changes I've made in case I've got something wrong.

hartenthaler commented:
The check is done only once in 24 hours, so you have to wait to see the message that there is a new version available in the control panel.

You have:
`public const CUSTOM_LATEST      = 'https://raw.githubusercontent.com/Neriderc/' . self::CUSTOM_MODULE. '/master/latest-version.txt';`
what is different to my example, but it should be the right one in your case (you have for example a branch "master" instead of "main" - there was some discussion to avoid the political incorrect master/slave words).

Neriderc commented:
I will wait for the 24 hours to see if it works.

The "master" branch was inherited from the previous repositories, but perhaps now is a good time to update terminology to the modern "main" terminology. I'll look at doing this.

Neriderc commented:
Right, it's working!

Also I've changed the default branch to "main" (and adjusted the URL here to match).

I've also added the support URL function so the link to the repo shows when a new version is out of date and shows when looking at the module page.
